core-plugins/docs/Projection-transform.md

```diff
@@ -32,21 +32,27 @@ and then field 'A' will be renamed to 'B':
     | B          | int  | 10     |
     +============================+
 
+Similarly, the transfrom will first check if it should keep a field, then rename it if configured to do so.
 
 Use Case
 --------
-The transform is used when you need to drop fields, change field types, or rename fields.
+The transform is used when you need to drop fields, keep specific fields, change field types, or rename fields.
 
 For example, you may want to rename a field from ``'timestamp'`` to ``'ts'`` because you want
-to write to a database where ``'timestamp'`` is a reserved keyword. Or, you might want to
+to write to a database where ``'timestamp'`` is a reserved keyword, or you might want to
 drop a field named ``'headers'`` because you know it is always empty for your particular
-data source. 
+data source. Or, you might want only keep fields named ``'ip'`` and ``'timestamp'`` and discard 
+all other fields.
 
 
 Properties
 ----------
 **drop:** Comma-separated list of fields to drop. For example: ``'field1,field2,field3'``.
 
+**keep:** Comma-seperated list of fields to keep. For example: ``'field1,field2,field3'``.
+
+Note: Both drop and keep fields can *not* be specified. At least one must be null or empty.
```

Comment:
Note: Drop and keep fields cannot *both* be specified. At least one must be either not defined, null, or an empty string.